examples/todoapp/bin/server/controllers/todos_controller.dart

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2023, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:shelf/shelf.dart';
+import 'package:shelf_router/shelf_router.dart';
+
+import '../services/todos_service.dart';
+import '../utils/api_controller.dart';
+import '../utils/server_exceptions.dart';
+
+class TodosController extends ApiController {
+  TodosController(this.todosService);
+
+  final TodosService todosService;
+
+  @override
+  void registerRequests(WrappedRouter router) {
+    router.addRequest(
+      RequestType.GET,
+      '/api/v1/todos',
+      getAllTodosHandler,
+    );
+    router.addRequest(
+      RequestType.POST,
+      '/api/v1/todos',
+      addTodoHandler,
+    );
+    router.addRequest(
+      RequestType.PUT,
+      '/api/v1/todos/<id>',
+      updateTodoByIdHandler,
+    );
+    router.addRequest(
+      RequestType.PATCH,
+      '/api/v1/todos/<id>',
+      updateCompletedByIdHandler,
+    );
+    router.addRequest(
+      RequestType.PATCH,
+      '/api/v1/todos',
+      updateCompletedForAll,
+    );
+    router.addRequest(
+      RequestType.DELETE,
+      '/api/v1/todos/completed',
+      deleteCompletedHandler,
+    );
+    router.addRequest(
+      RequestType.DELETE,
+      '/api/v1/todos/<id>',
+      deleteByIdHandler,
+    );
+    router.addRequest(
+      RequestType.GET,
+      '/api/v1/todos/<id>',
+      fetchTodoByIdHandler,
+    );
+  }
+
+  Future<Response> getAllTodosHandler(Request request) async {
+    final allTodos = todosService.fetchAllTodos();
+    return responseBuilder.buildOK(data: allTodos.toJson());
+  }
+
+  Future<Response> updateCompletedForAll(Request request) async {
+    final params = await request.bodyFromFormData();
+    final completed = params['completed'];
+    if (completed == null) {
```

Comment:
Is `throwIfEmpty` suitable for here?

Reply:
Unfortunately no, the  `void throwIfEmpty(String? param, Exception exception)` first argument of this function is of type String? and `completed` is bool